### PR TITLE
add create magic-link functionality

### PR DESCRIPTION
**Pull request by calhorn95**

### CODE REVIEW

**Comment by calhorn95** [comment]
SHOULD NOT BE MERGED UNTIL SMART LINKS MAKE IT TO MAIN

File: README.md [hunk 1]
```diff
@@ -151,3 +150,48 @@ deleted_user = psg.deleteUser(user_id)
...
+```python
+
+from passageidentity import Passage
+import os
+
+PASSAGE_APP_ID = os.environ.get("PASSAGE_APP_ID")
+PASSAGE_API_KEY = os.environ.get("PASSAGE_API_KEY")
+psg = Passage(PASSAGE_APP_ID, PASSAGE_API_KEY)
+
+# Get Passage User ID from database
```
Comment (dylanbrookes) [inline README.md:192]: Is this comment relevant to this example?

File: passageidentity/passage.py [hunk 2]
```diff
@@ -90,6 +104,39 @@ def authenticateJWT(self, token:str) -> Union[str, PassageError]:
...
+        send: bool
+        magic_link_path: str
+        redirect_url: str
+
+    """
+    Create Passage MagicLink
+    """
+    def createMagicLink(self, magicLinkAttributes: MagicLinkAttributes) -> Union[PassageMagicLinkType, PassageError]:
+        # if no api key, fail
+        if self.passage_apikey == "":
```
Comment (dylanbrookes) [inline passageidentity/passage.py:123]: If statement here can be refined I think (I know in Go we can't do this, but I'm pretty sure we can in Python):
```python
if !self.passage_apikey:
```

**Review by dylanbrookes — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (calhorn95) [reply README.md:192]: nope
